### PR TITLE
[DOC] Transformers tutorial

### LINKED ISSUE #1025: [DOC] transformers tutorial
There should be a notebook that explains the different transformer scitypes, and how transformers work in `sktime`.

### PR DESCRIPTION
**Pull request by fkiraly**
This resolves #1025 by providing a comprehensive introduction to the transformers in `sktime`.

The introduction currently includes the first chapter which explains the key transformer types, their signature, and how users can search for transformers.

The sections with further details are currently WiP, but I think this is good enough as a first pass and helpful to users.

This PR also adds one fixture for a univariate, positive-valued series, to have an easy example to showcase the `BoxCoxTransformer` (which requires positive values).

### CODE REVIEW

**Review by Lovkush-A — COMMENTED** [review]
Had an initial skim through the notebook. Overall looks good. Plenty of examples, which I think cover the main options.

Detailed suggestions to come later in the week, but some things I noticed for now:
* need to remove references to forecasting, that have stuck around from copying forecasating notebook.
* need to update the contents list
* we want to come up with consistent way of naming 'normal transformers' to distinguish from 'pairwise transformers'. Some suggestions: unary transformers, standard transformers, transformers.  If you go for the final option (just 'transformers'), then maybe the top title in the notebook should be 'Using sktime transformers and pairwise transformers'

**Comment by fkiraly** [comment]
@Lovkush-A, I've updated the table of contents, removed references to forecasting, updated terminology.

File: examples/05_transformers.ipynb [hunk 1]
```diff
@@ -0,0 +1,851 @@
...
+    "\n",
+    "`Series` are commonly represented as:\n",
+    "\n",
+    "* `pandas.Series` for univariate time series and sequences\n",
+    "* `pandas.DataFrame` for uni- or multivariate time series and sequences\n",
+    "\n",
+    "The `Series.index` and `DataFrame.index` are used for representing the time series or sequence index. `sktime` supports pandas integer, period and timestamp indices.\n",
+    "\n",
+    "`Panel`-s are commonly represented as:\n",
+    "* a `pandas.DataFrame` in a specific format, defined by the `pd-multiindex` scitype - this has a double row index, for time points and instances\n",
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:120]: * simplifying sentence structure, by creating two sentences.
* I don't think 'double row index' is correct phrasing. I checked pandas and I there is variable 'nlevels', hence I am suggesting '2-level'

```suggestion
    "* a `pandas.DataFrame` in a specific format, defined by the `pd-multiindex` mtype. This has a 2-level index, for time points and instances\n",
```

File: examples/05_transformers.ipynb [hunk 2]
```diff
@@ -0,0 +1,851 @@
...
+    "* how to search `sktime` for transformers of a certain type"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.1 Data container format<a class=\"anchor\" id=\"section_1_1\"></a>\n",
+    "\n",
+    "`sktime` transformers apply to individual time series and panels (= collections of time series).\n",
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:107]: Adding explanation of the term 'instance' that we use later on, and re-writing the stuff in paranthesis as a sentence.

```suggestion
    "`sktime` transformers apply to individual time series and panels. Panels are collections of time series, and we refer to each time series in a panel as an \"instance\" of the panel.\n",
```

File: examples/05_transformers.ipynb [hunk 3]
```diff
@@ -0,0 +1,851 @@
...
+    "get_examples(\"pd.DataFrame\", \"Series\")[1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# example of a panel in pd-multiindex format\n",
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:168]: maybe use the term mtype more throughout the notebook, so a user gets used to the term. I think this is important because we *do* use the word mtype as a parameter in many of these functions, so it is good for a user to be exposed to that word more in the tutorial.

```suggestion
    "# example of a panel with mtype pd-multiindex\n",
```

File: examples/05_transformers.ipynb [hunk 4]
```diff
@@ -0,0 +1,851 @@
...
+    "get_examples(\"pd-multiindex\", \"Panel\")[0]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# example of the same panel in df-list format\n",
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:178]: If you agre with suggestion above
```suggestion
    "# example of the same panel with mtype df-list\n",
```

File: examples/05_transformers.ipynb [hunk 5]
```diff
@@ -0,0 +1,851 @@
...
+    "get_examples(\"df-list\", \"Panel\")[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`sktime` supports more in-memory formats, see the \"datatypes and datasets\" tutorial for more details."
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:186]: if you agree with suggestion above
```suggestion
    "`sktime` supports more mtypes, see the \"datatypes and datasets\" tutorial for more details."
```

File: examples/05_transformers.ipynb [hunk 6]
```diff
@@ -0,0 +1,851 @@
...
+   "metadata": {},
+   "source": [
+    "Transformers for `Series` and `Panel` have the same high-level interface. Depending which data type they are more commonly used for, they are found either in the `transformations.series` or `transformations.panel` module. As said, this does not imply a separate interface.\n",
+    "\n",
+    "The most important interface points of transformers are:\n",
+    "\n",
+    "1. construction with parameters, this is as with any other `sktime` estimator\n",
+    "2. fitting the transformer, via `fit`\n",
+    "3. transforming data, via `transform`\n",
+    "4. inverse transforming, via `inverse_transform`\n",
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:207]: maybe comment that not all transformers have inverse_transform?

File: examples/05_transformers.ipynb [hunk 7]
```diff
@@ -0,0 +1,851 @@
...
+    "5. updating the transformer, via `update` - not all transformers have this interface point (`update` is currently work in progress, as of v0.8.x, contributions are appreciated)\n",
+    "\n",
+    "We show this in two examples below.\n",
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:210]:
```suggestion
    "We show this in two example transformers below: one series-to-series transformer and one series-to-primitive.\n",
```

File: examples/05_transformers.ipynb [hunk 8]
```diff
@@ -0,0 +1,851 @@
...
+    "\n",
+    "We will apply transformations to the following `Series` and `Panel` data:"
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:212]: making clear that 'transformations' refers to the examples in the previous sentence.
```suggestion
    "We will apply both transformations to the following `Series` and `Panel` data:"
```

File: examples/05_transformers.ipynb [hunk 9]
```diff
@@ -0,0 +1,851 @@
...
+   "outputs": [],
+   "source": [
+    "from sktime.datatypes import get_examples\n",
+    "\n",
+    "# univariate series used in the examples\n",
+    "X_series = get_examples(\"pd.Series\", \"Series\")[0][0:3]\n",
+    "# panel used in the examples\n",
+    "X_panel = get_examples(\"pd-multiindex\", \"Panel\")[0][[\"var_1\"]]\n",
+    "# sub-setting is needed since the example transformer (box-cox)\n",
+    "#   wants univariate and positive data"
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:228]: maybe use an example that does not need sub-setting, to avoid having to explain this?

File: examples/05_transformers.ipynb [hunk 10]
```diff
@@ -0,0 +1,851 @@
...
+   "source": [
+    "# transforming the series\n",
+    "my_boxcox_trafo.transform(X_series)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Generally, the series passed to `transform` need not be the same as in `fit` - if it is, the shorthand `fit_transform` can instead be used:"
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:308]: suggestion for slightly clearer wording
```suggestion
    "Generally, the series passed to `transform` need not be the same as in `fit`, but if they are the same, the shorthand `fit_transform` can instead be used:"
```

File: examples/05_transformers.ipynb [hunk 11]
```diff
@@ -0,0 +1,851 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As before, we can fit/apply with `fit`, `transform`, and `fit_transform`.\n",
+    "\n",
+    "`SummaryTransformer` returns primitive features, hence the output will be a `pandas.DataFrame`, each row corresponding to one series in the input. \n",
+    "\n",
+    "If the input is a single series, the output of `transform` and `fit_transform` will be a one-row `DataFrame`, corresponding to mean etc of that one series:"
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:365]:
```suggestion
    "If the input is a single series, the output of `transform` and `fit_transform` will be a one-row `DataFrame`, corresponding to the summary statistics of that one series:"
```

File: examples/05_transformers.ipynb [hunk 12]
```diff
@@ -0,0 +1,851 @@
...
+   "outputs": [],
+   "source": [
+    "my_summary_trafo.fit_transform(X_series)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If the input is a panel, the output of `transform` and `fit_transform` will be a `DataFrame` with as many rows as the `Panel` had series. The row with the corresponding index has mean etc of the series in the panel `X_panel` with the same index:"
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:381]: * replacing 'mean etc' as in suggestion above
* using `i`th instead of 'corresponding index'. I think this is clearer
```suggestion
    "If the input is a panel, the output of `transform` and `fit_transform` will be a `DataFrame` with as many rows as the `Panel` had series. The `i`th row has the summary statistics of the `i`th series in the panel `X_panel`:"
```

File: examples/05_transformers.ipynb [hunk 13]
```diff
@@ -0,0 +1,851 @@
...
+   "outputs": [],
+   "source": [
+    "my_summary_trafo.fit_transform(X_panel)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### transformers with series/panel output vs primitive output\n",
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:397]:
```suggestion
    "#### Transformers with series/panel output vs primitive output\n",
```

File: examples/05_transformers.ipynb [hunk 14]
```diff
@@ -0,0 +1,851 @@
...
+    "\n",
+    "Whether a transformer will return primitives, i.e., a `pandas.DataFrame` in `transform`, or time series like objects (`Series` or `Panel`), can be checked by using the `\"scitype:transform-output\"` tag. This is `\"Series\"` for behaviour as in the first example (box-cox), and `\"Primitives\"` for behaviour as in the second example (summarizer):"
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:399]: * minor restructuring of first sentence to hopefully make clearer, e.g. putting 'ie a dataframe' into brackets, reordering so that time series comes before primitives (to match the second sentence)
* explicitly naming the example transformers used
```suggestion
    "Whether `transform` will return time series like objects (`Series` or `Panel`) or primitives (i.e. a `pandas.DataFrame`) can be checked by using the `\"scitype:transform-output\"` tag. This is `\"Series\"` for behaviour as in the first example (`BoxCoxTransformer`), and `\"Primitives\"` for behaviour as in the second example (`SummaryTransformer`):"
```

File: examples/05_transformers.ipynb [hunk 15]
```diff
@@ -0,0 +1,851 @@
...
+   "metadata": {},
+   "source": [
+    "Use of tags to characterize and search for transformers will be discussed in more detail in section 4."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NOTE: currently not all transformers are refactored to the full interface that provides vectorization from `Series` to `Panel` data, the above may hence not fully work for all transformers. Contributions to the transformer refactor are very much appreciated."
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:431]: this is the first time vectorization is mentioned, so this will confuse some users. We should indicate precisely which parts of the above relied on vectorisation, e.g. we should explain that vectorization is occuring when applying box-cox to a panel.

**Review by Lovkush-A — CHANGES_REQUESTED** [review]
detailed comments up to but not including section 1.3. All minor suggestions (little rewordings, typos, etc.)

File: examples/05_transformers.ipynb [hunk 16]
```diff
@@ -0,0 +1,851 @@
...
+   "source": [
+    "## Table of Contents\n",
+    "\n",
+    "* [1. Basic forecasting workflows](#chapter1)\n",
+    "    * [1.1 Data container format](#section_1_1)\n",
+    "    * [1.2 General transformer signature - simple transformers](#section_1_2)   \n",
+    "    * [1.3 General transformer signature - pairwise series transformers](#section_1_3)         \n",
+    "    * [1.4 General transformer signature - pairwise transformers](#section_1_4) \n",
+    "    * [1.5 Searching for transformers](#section_1_5)         \n",
+    "* [2. Time series transformers in detail](#chapter2)\n",
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply examples/05_transformers.ipynb:58]: made it consistent, but changed it to the other option - "time series transformer", to allow disambiguation from `scikit-learn` transformers

File: examples/05_transformers.ipynb [hunk 17]
```diff
@@ -0,0 +1,851 @@
...
+    "\n",
+    "Transformers are modularized data processing steps commonly used in machine learning. `sktime` introduces transformer types related to time series, which can be used in pipeline constructs across learning tasks. For instance, the same time series feature extractor could be used in a forecasting pipeline, or a time series classification pipeline.\n",
+    "\n",
+    "Note: the term \"transformer\" is used in the sence of `scikit-learn`. This should be distinguished from the \"transformer\" referring to specific neural network architecture.\n",
+    "\n",
+    "`sktime` provides common interfaces to different types of transformations for stand-alone or composition use. As all `sktime` estimators, transformers expose parameters which can be tuned inside pipelines and composites.\n",
+    "\n",
+    "**Section 1** provides an overview of the different kinds of transformers in `sktime`.\n",
+    "\n",
+    "**Section 2** discusses in more detail time series and panel transformers, and common uses in pipelines.\n",
```
[the remark this answers is not in the document]
Comment (fkiraly) [reply examples/05_transformers.ipynb:39]: see above, changed to "time series transformer" consistently

File: examples/05_transformers.ipynb (hunk 11 above)
Comment (fkiraly) [reply examples/05_transformers.ipynb:365]: clarified this, but differently, since not everyone may be able to visualize what "summary statistics" means here

File: examples/05_transformers.ipynb (hunk 15 above)
Comment (fkiraly) [reply examples/05_transformers.ipynb]: clarified this by removing the reference

File: examples/05_transformers.ipynb (hunk 16 above)
[the remark this answers is not in the document]
Comment (Lovkush-A) [reply examples/05_transformers.ipynb:58]: works for me!

File: examples/05_transformers.ipynb [hunk 18]
```diff
@@ -0,0 +1,851 @@
...
+    "This section explains the differrent types of transformers found in `sktime`.\n",
+    "\n",
+    "There are four main types of transformation in `sktime`:\n",
+    "\n",
+    "* transforming a series/sequence into scalar- or category-valued features. Examples: `tsfresh`, or extracting `mean` and `variance` overall.\n",
+    "* transforming a series into another series. Examples: detrending, smoothing, filtering, lagging.\n",
+    "* transforming a panel into another panel. Examples: principal component projection; applying individual series transformation to all series in the panel.\n",
+    "* transforming a pair of series into a scalar value. Examples: dynamic time warping distance between series/sequences; generalized alignment kernel between series/sequences.\n",
+    "\n",
+    "Notably, the first three (series to primitive features, series to series, panel to panel) are covered by the same base class template and module. We call these transformers \"simple transformers\", or, simply, \"transformers.\n",
```
[the remark this answers is not in the document]
Comment (Lovkush-A) [reply examples/05_transformers.ipynb]: @fkiraly. This suggestion still needs implementing (missing quote at the end of the line)

File: examples/05_transformers.ipynb [hunk 19]
```diff
@@ -0,0 +1,851 @@
...
+    "Kernels and distances for time series and sequences have the same mathematical signature and differ only in mathematical properties (e.g., definiteness assumptions) - they are covered by the more abstract scientific type of \"pairwise transformer\".\n",
+    "\n",
+    "Below, we give an overview in sub-sections:\n",
+    "* reviewing common data container formats for series and panels\n",
+    "* showcasing the signature of transformers that transform a series or panel\n",
+    "* showcasing the signature of transformers that transform a pair ot series to a scalar, e.g., distances or kernels\n",
```
[the remark this answers is not in the document]
Comment (Lovkush-A) [reply examples/05_transformers.ipynb:97]: @fkiraly. this conversation still to be resolved

**Comment by fkiraly** [comment]
apologies for missing some comments, my browser sometimes seems to eat them.

File: examples/05_transformers.ipynb (hunk 9 above)
Comment (fkiraly) [reply examples/05_transformers.ipynb:228]: yes, I added a fixture just for that recently. Good reminder.

File: examples/05_transformers.ipynb (hunk 7 above)
Comment (fkiraly) [reply examples/05_transformers.ipynb:210]: these terms have not been introduced, so I used a different explanation

File: examples/05_transformers.ipynb (hunk 9 above)
Comment (fkiraly) [reply examples/05_transformers.ipynb:228]: ... didn't work since needs univariate. Added a positive fixture now.

File: examples/05_transformers.ipynb (hunk 6 above)
Comment (fkiraly) [reply examples/05_transformers.ipynb:207]: added

File: examples/05_transformers.ipynb (hunk 18 above)
[the remark this answers is not in the document]
Comment (fkiraly) [reply examples/05_transformers.ipynb]: done

File: examples/05_transformers.ipynb [hunk 20]
```diff
@@ -0,0 +1,849 @@
...
+   "source": [
+    "### 1.4 General transformer signature - pairwise transformers<a class=\"anchor\" id=\"section_1_4\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`sktime` also provides functionality for pairwise transformers on tabular data, i.e., mathematical objects of signature `(DataFrame-row, DataFrame-row) -> float`, or, in mathematical notation, $$\\mathbb{R}^n \\times\\mathbb{R}^n\\rightarrow\\mathbb{R}$$.\n",
+    "Common examples are distances between series, or (positive definite) kernels on series.\n",
```
Comment (Lovkush-A) [inline examples/05_transformers.ipynb:579]: i think this line needs to be udpated; curerntly it is just copied from section 1.3
```suggestion
    "Common examples are distances between series, or (positive definite) kernels on series.\n",
```

**Review by Lovkush-A — CHANGES_REQUESTED** [review]
* Happy with all the updates since my last review
* finished reading through the remainder of the notebook and only saw one small thing to improve
* one observation is that the name of the scitypes (`transformer`, `transformer-pairwise` and `transformer-panel` ) do not match well with the naming used in the notebook. Maybe need to update the names of the scitypes in a future PR?

**Comment by lmmentel** [comment]
I notices that all the link in the TOC are resolved to the repo folder with examples, is that intentional?

File: examples/05_transformers.ipynb [hunk 21]
```diff
@@ -0,0 +1,849 @@
...
+    "* [3. Pairwise transformers in detail](#chapter3)  \n",
+    "* [4. Extension guide - implementing your own transformer](#chapter4)        \n",
+    "* [5. Summary](#chapter5)          "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### package imports"
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:68]: no imports directly following this cell

File: examples/05_transformers.ipynb [hunk 22]
```diff
@@ -0,0 +1,849 @@
...
+    "## 1. Overview of transformers in `sktime` <a class=\"anchor\" id=\"chapter1\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This section explains the different types of transformers found in `sktime`.\n",
+    "\n",
+    "There are four main types of transformation in `sktime`:\n",
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:84]: Would it be possible to include the names or classes/mixins of the different transformers before the explanations in the list below?

File: examples/05_transformers.ipynb [hunk 23]
```diff
@@ -0,0 +1,849 @@
...
+    "`Series` are commonly represented as:\n",
+    "\n",
+    "* `pandas.Series` for univariate time series and sequences\n",
+    "* `pandas.DataFrame` for uni- or multivariate time series and sequences\n",
+    "\n",
+    "The `Series.index` and `DataFrame.index` are used for representing the time series or sequence index. `sktime` supports pandas integer, period and timestamp indices.\n",
+    "\n",
+    "`Panel`-s are commonly represented as:\n",
+    "* a `pandas.DataFrame` in a specific format, defined by the `pd-multiindex` mtype. This has a 2-level index, for time points and instances\n",
+    "* a `list` of `pandas.DataFrame`, where all `pandas.DataFrame` are in the `Series` format. The different `list` elements are the different instances\n",
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:121]: This is rather unclear since `Series` is ambiguously used, also it is unclear what a `DataFrame` is the format of `Series` would be or how to convert it.

File: examples/05_transformers.ipynb [hunk 24]
```diff
@@ -0,0 +1,849 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.1 Data container format<a class=\"anchor\" id=\"section_1_1\"></a>\n",
+    "\n",
+    "`sktime` transformers apply to individual time series and panels. Panels are collections of time series, and we refer to each time series in a panel as an \"instance\" of the panel.\n",
+    "This is formalized as abstract \"scientific types\" `Series` and `Panel`, with multiple possible in-memory representations, so-called \"mtypes\".\n",
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:108]: is it possible to link to documentation definitions of `Series` and `Panel` ?

File: examples/05_transformers.ipynb [hunk 25]
```diff
@@ -0,0 +1,849 @@
...
+    "\n",
+    "For the purpose of this tutorial, we will be working with the most common mtypes. For more details and formal data type specifications, see the \"datatypes and datasets\" tutorial.\n",
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:110]: Does this tutorial exist, if so let's add a link to it. If it doesn't maybe it better not o mention it.

File: examples/05_transformers.ipynb [hunk 26]
```diff
@@ -0,0 +1,849 @@
...
+    "* how to search `sktime` for transformers of a certain type"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.1 Data container format<a class=\"anchor\" id=\"section_1_1\"></a>\n",
+    "\n",
+    "`sktime` transformers apply to individual time series and panels. Panels are collections of time series, and we refer to each time series in a panel as an \"instance\" of the panel.\n",
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:107]: It is unclear what is the relation of generic term `time series` to sci types and mtypes so here `Panels are collections of time series,` is a rather vague statement.

File: examples/05_transformers.ipynb [hunk 27]
```diff
@@ -0,0 +1,849 @@
...
+    "This is formalized as abstract \"scientific types\" `Series` and `Panel`, with multiple possible in-memory representations, so-called \"mtypes\".\n",
+    "\n",
+    "For the purpose of this tutorial, we will be working with the most common mtypes. For more details and formal data type specifications, see the \"datatypes and datasets\" tutorial.\n",
+    "\n",
+    "`Series` are commonly represented as:\n",
+    "\n",
+    "* `pandas.Series` for univariate time series and sequences\n",
+    "* `pandas.DataFrame` for uni- or multivariate time series and sequences\n",
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:115]: Series are represented as Series or a DataFrame... this is naming collision. If there is no other word that can be used here is should at least be formatted differntly and possible explained earlier that there is a semantic difference between `Series` and `pandas.Series`

File: examples/05_transformers.ipynb [hunk 28]
```diff
@@ -0,0 +1,849 @@
...
+    "\n",
+    "The `Series.index` and `DataFrame.index` are used for representing the time series or sequence index. `sktime` supports pandas integer, period and timestamp indices.\n",
+    "\n",
+    "`Panel`-s are commonly represented as:\n",
+    "* a `pandas.DataFrame` in a specific format, defined by the `pd-multiindex` mtype. This has a 2-level index, for time points and instances\n",
+    "* a `list` of `pandas.DataFrame`, where all `pandas.DataFrame` are in the `Series` format. The different `list` elements are the different instances\n",
+    "\n",
+    "In either case, the \"time\" index must be a `sktime` compatible time index type, as for `Series`."
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:123]: This requires elaboration.

File: examples/05_transformers.ipynb [hunk 29]
```diff
@@ -0,0 +1,849 @@
...
+   "metadata": {},
+   "source": [
+    "### 1.2 General transformer signature - time series transformers<a class=\"anchor\" id=\"section_1_2\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Transformers for `Series` and `Panel` have the same high-level interface. Depending which data type they are more commonly used for, they are found either in the `transformations.series` or `transformations.panel` module. As said, this does not imply a separate interface.\n",
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:200]: Is the intention here to say that the interface is the same or that there are two separate modules?

File: examples/05_transformers.ipynb [hunk 30]
```diff
@@ -0,0 +1,849 @@
...
+    "\n",
+    "The most important interface points of transformers are:\n",
+    "\n",
+    "1. construction with parameters, this is as with any other `sktime` estimator\n",
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:204]: maybe it's better to show not tell here. The second part seems like an irrelevant implementation detail to mention.

File: examples/05_transformers.ipynb [hunk 31]
```diff
@@ -0,0 +1,849 @@
...
+    "2. fitting the transformer, via `fit`\n",
```
Comment (lmmentel) [inline examples/05_transformers.ipynb:205]: I would reverse the order of definitions here going for `method`: description of what it does, possibly a signature. Sth along the lines of:

- `fit` fitting of the transformer parameters, accepts `(X, y)` ...

**Comment by fkiraly** [comment]
@lmmentel, I think your points are hopefully addressed by the new first section which is meant as a lightweight intro.

In prep for pydata global, I will merge this in the next days, based on @Lovkush-A's feedback, time constraints, and since this is an improvement above "no tutorial".

**Review by aiwalter — APPROVED** [review]
